checker/src/main/java/org/checkerframework/checker/nullness/NullnessNoInitAnnotatedTypeFactory.java

```diff
@@ -113,6 +113,9 @@ public class NullnessNoInitAnnotatedTypeFactory
     private final ExecutableElement mapGet =
             TreeUtils.getMethod("java.util.Map", "get", 1, processingEnv);
 
+    // High-level optimization: cache entire addComputedTypeAnnotations processing flow
```

Comment:
The Javadocs should document what is being cached, in particular if the key/value is a String. What format is the String in?
For a cache you would also expect to see the value it is mapped to, so why is a Set enough?

Who cleans up this cache? There are many allocations in large programs and we don't want a Set that becomes arbitrarily large.
Maybe it should be cleared every time a new root is set? See [here](https://github.com/eisop/checker-framework/blob/448f0d6f3b37a08995c88757afb1b64624697bbe/framework/src/main/java/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java#L469).